src/main/java/com/github/cameltooling/lsp/internal/completion/CamelOptionValuesCompletionsFuture.java

```diff
@@ -88,6 +89,22 @@ public List<CompletionItem> apply(CamelCatalog camelCatalog) {
 					items.add(langItem);
 				}
 				return items;
+			} else if(StringUtils.startsWithIgnoreCase(
```

Comment:
the apply method is starting to be quite long. i'm wondering if it is not time to extract methods/classes for each cases.

And realizing that Sonar is starting to raise a warning on the complexity

Reply:
Refactoring to do it in a separated function for all "check based on the parameter value" cases.